File: code/csharp/DesignPatterns/iterator/IteratorClient.cs
using System;

namespace Iterator
{
    public class IteratorClient
    {
        public static void Iterator()
        {
            IBlog blog = new BlogPlatform();
            blog.AddPost(new Post("Title 1", "Author 1", "Summary 1"));
            blog.AddPost(new Post("Title 2", "Author 2", "Summary 2"));

            IIterator iterator = blog.CreateIterator();
            while (iterator.HasNext())
            {
                Post post = iterator.Next();
                Console.WriteLine($"Title: {post.Title}, Author: {post.Author}, Summary: {post.Summary}");
            }
        }
    }
}

File: code/typescript/iterator/IteratorClient.ts
import { BlogPlatform } from "./BlogPlatform";
import { Post } from "./Post";

export function Iterator(): void {
    let blog: BlogPlatform = new BlogPlatform();
    blog.addPost(new Post("Title 1", "Author 1", "Summary 1"));
    blog.addPost(new Post("Title 2", "Author 2", "Summary 2"));

    let iterator = blog.createIterator();
    while (iterator.hasNext()) {
        let post = iterator.next();
        console.log(`Title: ${post.title}, Author: ${post.author}, Summary: ${post.summary}`);
    }
}
